### PR TITLE
Balder/mandatory error check in add

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 1]
```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
...
+        return (rlen == 0)
+               ? true
+               : handleReadError("packet length", file, sizeof(len), rlen, lastKnownGoodPos, allowTruncate);
+    }
+
+    try {
+        chunk = IChunk::create(encoding);
+    } catch (const std::exception & e) {
+        string msg(make_string("Version mismatch. Expected 'ccitt_crc32=1' or 'xxh64=2',"
+                               " got %d from '%s' at position %ld",
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:573]: PRId64 is the correct format string for position.

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 2]
```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
...
+    }
+    rlen = file.Read(buf.get(), len);
+    if (rlen != len) {
+        return handleReadError("packet blob", file, len, rlen, lastKnownGoodPos, allowTruncate);
+    }
+    try {
+        nbostream_longlivedbuf is(buf.get(), len);
+        chunk->decode(is);
+    } catch (const std::exception & e) {
+        throw runtime_error(make_string("Got exception during decoding of packet '%s' from file '%s' (len pos=%" PRId64 ", len=%d)",
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:593]: len pos ?

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 3]
```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
...
-    os << crc;
-    size_t osSize = os.size();
-    assert(osSize == len + sizeof(len) + sizeof(uint8_t));
-
+    size_t begin = os.wp();
+    os << _encoding.getRaw();
+    os << uint32_t(0);
+    Encoding realEncoding = chunk.encode(os);
+    size_t end = os.wp();
+    os.wp(0);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:540]: Consider adding comment about patching up header of serialized chunk.

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 4]
```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
...
-    size_t start(os.size());
-    entry.serialize(os);
-    size_t end(os.size());
-    crc = calcCrc(_encoding.getCrc(), os.data() + start, end - start);
-    os << crc;
-    size_t osSize = os.size();
-    assert(osSize == len + sizeof(len) + sizeof(uint8_t));
-
+    size_t begin = os.wp();
+    os << _encoding.getRaw();
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:536]: Consider adding comment about this being a placeholder value, to be patched up later.

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:573]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:593]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:540]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 4 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:536]: Fixed

**Pull request merged**